devicetypes/smartthings/zwave-fan-controller.src/zwave-fan-controller.groovy

```diff
@@ -108,21 +108,33 @@ def zwaveEvent(physicalgraph.zwave.commands.hailv1.Hail cmd) {
 
 def fanEvents(physicalgraph.zwave.Command cmd) {
 	def value = (cmd.value ? "on" : "off")
-	def result = [createEvent(name: "switch", value: value)]
-	result << createEvent(name: "level", value: cmd.value == 99 ? 100 : cmd.value)
+	def result = [createEvent(name: "switch", value: value),
+				  createEvent(name: "level", value: cmd.value == 99 ? 100 : cmd.value)]
+
 	def fan_level = Math.ceil(cmd.value/33) as int
-//	if (cmd.value < 33 && cmd.value >= 1) fan_level = 1 //sometimes we get "1" when the device is on low
+
+	// The GE, Honeywell, and Leviton treat 33 as medium, so account for that
+	if (cmd.value == 33) {
```

Comment:
What other values do they report? I just wonder if it would be better to change the code above to do `Math.ceil((cmd.value + 1)/33) as int`

Reply:
Have never been able to get a consistent answer about that, hence all the changes to this DTH

Reply:
1-32 = 1,
33-66 = 2,
67 - 99 or 100 = 3

Bashi realized that calling setLevel could be used as an indicator as to what that value means to the bulb.

Reply:
I suspect the firmware does a check on the values and not some math.

if 1 <= i <= 32

else if 33 <= i <= 66

else

🤷‍♂

Reply:
Personally I think the range checks would be preferable over the use of ceil and the special check for 33

Reply:
@tpmanley I am going to give this a whirl today. Made some updates.